src/libraries/System.ComponentModel.Annotations/src/System.ComponentModel.Annotations.csproj

```diff
@@ -2,6 +2,8 @@
   <PropertyGroup>
     <TargetFrameworks>netstandard2.1</TargetFrameworks>
     <Nullable>enable</Nullable>
+    <!-- Since many resource strings are shown to an end-user, opt out of resource substitutions in DataAnnotations. -->
+    <GenerateResourcesSubstitutions>false</GenerateResourcesSubstitutions>
```

Comment:
If we do it this way, then ResourceManager class would be preserved right? Wouldn't it be better to add defaults to all the places where the resources are actually used like we did in StackTrace? I know that might be ugly but we should make sure we don't regress size

Reply:
This is a good point. I'm a little bit torn though. 

`System.ComponentModel.Annotations` isn't used by default. So an app would have to opt in to using DataAnnotations in order for ResourceManager to be preserved. At some point, ResourceManager is going to get included by someone anyway though. For example, I looked at the [BlazingPizza App](https://github.com/dotnet-presentations/blazor-workshop), and it references ResourceManager through [Microsoft.AspNetCore.Authorization.Infrastructure](https://github.com/dotnet/aspnetcore/blob/fe1a56972e8ff247d8b000448ea4ccc0b17ec833/src/Security/Authorization/Core/src/RolesAuthorizationRequirement.cs#L30). So in apps like that, ResourceManager will be there anyway (same for any other 3rd party library that uses a .resx for exception messages).

Doing a quick analysis - of the 67 references to `SR.` in this project, 35 of them are to `throw` exceptions. I assume the other 32 of them are for validation messages of some sort or another. That is a lot of places to hard code the messages in the code.

On one hand it is great to limit the usages of ResourceManager, but if an app is already using ResourceManager, limiting the usage of it doesn't save much.

Thoughts?

Reply:
I don't have a strong preference as I'm not familiar how common it is to use DataAnnotations for filling forms on Blazor apps,perhaps @pranavkm can share how common it is?
If it's very common, then we might want to consider doing it in those 35 places, since even though the price would be uglier and harder to maintain code, the benefits of trimming resource manager for common apps would be probably worth it. All that said, I'm happy with whichever call you decide to make.

Reply:
A comment from @pranavkm on the issue this is fixing:
> Forms and validation is pretty essential to Blazor and our built-in Component uses DataAnnotations. It would be frustrating for developers to have to remember to modify their project file any time they use validation. I'd certainly ask.

That suggests this is pretty common so might be worth considering it given all the work that was put to be able to trim ResourceManager.

Reply:
In 5.0 we did only the first step on the size reduction journey and fixing it this way only means we'll have to fix it better next time. I think we should fix it properly if not in 5.0 then in master so when someone sets `UseSystemResourceKeys` it does the expected thing.

Reply:
What's the "proper fix" here?

Reply:
@eerhardt one option could be

1. Fix the keys to be more descriptive. In the reported case, the key is almost as long as the text and yet it's very ambiguous
2. Make it possible for developers to disable `UseSystemResourceKeys` for individual assemblies if they decide to prefer verbose output over app size

Reply:
I've updated the approach taken here. See https://github.com/dotnet/runtime/pull/42274#issuecomment-696414182.